### PR TITLE
fix: v-markdown directive for SSR

### PR DESCRIPTION
**Pull request by Samitier**
We have a problem with `v-markdown` directive 🔥

Looks like nuxt doesn't render directives server-side [unless we specify stuff in `nuxt.config.js`](https://github.com/nuxt/nuxt.js/issues/898). This means that all content set with `v-markdown` is not in the SSR page and it makes some popping when the app hydrates because the content [appears after the page loads](https://www.holaluz.com/autoconsumo/). We haven't had any trouble with SEO, everything indexes correctly, but I guess its better to not risk it.

We should add this to every `nuxt.config` of the SSR projects that use `v-markdown`:

```js
import { markdown } from '@holaluz/margarita'

[...]

render: {
    bundleRenderer: {
      directives: {
        markdown(vnode, directiveMeta) {
          vnode.data.domProps = {
            ...vnode.data.domProps,
            childNodes: [],
          }
          markdown(vnode.data.domProps, directiveMeta)
        },
      },
    },
  },
```
In this PR I added the SSR implementation of `v-markdown` directly in margarita so we can do something like this:

```js
import { markdownSSR } from '@holaluz/margarita'

[...]

render: {
    bundleRenderer: {
      directives: {
         markdown: markdownSSR
      }
    }
}
```

### CODE REVIEW

**Comment by afontcu** [comment]
I vote for the 3rd solution, too! However I'd not introduce if statements in unit tests 😬

**Comment by Samitier** [comment]
> I vote for the 3rd solution, too! However I'd not introduce if statements in unit tests 😬

Added markdown SSR & fixed the tests

File: src/index.spec.js [hunk 1]
```diff
@@ -4,12 +4,14 @@ import isVueComponent from 'is-vue-component'
...
 import * as Margarita from './index'
 
-// Get the default export...
+// Get the default export
 const defaultExport = Margarita.default
 
-// ...and remove it from the object, so we have the list of imported comps.
-// eslint-disable-next-line no-import-assign
-delete Margarita.default
+// Remove default export and directives, so we have the list of imported comps.
```
Comment (afontcu) [inline src/index.spec.js:10]: neat 👌

**Comment by afontcu** [comment]
Could we add the config snippet in the codebase, or in the story, or somewhere? To make it easy for everyone to find the right Nuxt config for markdown SSR 👌

**Comment by Samitier** [comment]
> Could we add the config snippet in the codebase, or in the story, or somewhere? To make it easy for everyone to find the right Nuxt config for markdown SSR 👌

image

I added this. What do you think?

**Review by afontcu — APPROVED** [review]
neat 👌

**Pull request merged**